src/FolderContext.ts

```diff
@@ -29,7 +30,6 @@ export class FolderContext implements vscode.Disposable {
     ) {
         if (this.isRootFolder) {
             this.packageWatcher = new PackageWatcher(this, workspaceContext);
-            this.packageWatcher.install();
```

Comment:
What does this have to do with weak references?

Reply:
The install and constructor have been combined in this PR. I guess it isn't completely related